internal/cmd/load-balancer/generate-payload/generate_payload.go

```diff
@@ -152,13 +150,37 @@ func NewCmd(p *print.Printer) *cobra.Command {
 			if err != nil {
 				return fmt.Errorf("read load balancer: %w", err)
 			}
+
+			listeners := *resp.Listeners
+
+			for i := range listeners {
+				listener := listeners[i]
+				listeners[i] = loadbalancer.Listener{
+					DisplayName: listener.DisplayName,
+					Port:        listener.Port,
+					Protocol:    listener.Protocol,
+					TargetPool:  listener.TargetPool,
+				}
+
+				if listener.ServerNameIndicators != nil {
+					listeners[i].ServerNameIndicators = listener.ServerNameIndicators
+				}
+
+				if listener.Tcp != nil {
+					listeners[i].Tcp = listener.Tcp
+				}
+
+				if listener.Udp != nil {
+					listeners[i].Udp = listener.Udp
+				}
+			}
+
 			updatePayload := &loadbalancer.UpdateLoadBalancerPayload{
 				ExternalAddress: resp.ExternalAddress,
-				Listeners:       resp.Listeners,
+				Listeners:       &listeners,
 				Name:            resp.Name,
 				Networks:        resp.Networks,
 				Options:         resp.Options,
-				PrivateAddress:  resp.PrivateAddress,
 				TargetPools:     resp.TargetPools,
 				Version:         resp.Version,
 			}
```

Comment:
I would abstract this logic in a function that can be unit tested

Reply:
Also, can't we just set the invalid fields to `nil` instead of recreating each listener?

Reply:
Implemented setting listener.name to nil in a new function, added unit test.